### PR TITLE
feat(model-types): add generated modeltypes

### PR DESCRIPTION
**Pull request by Birkbjo**
Generated from `OpenAPI`-schemas using https://github.com/Birkbjo/dhis2-open-api-ts

Note that this included all models/components from the openApiSchema. Some unused stuff are filtered out by the tool above.
However there are models we will not use in the maintenance app, but I'm not sure if it's worth filtering those out. 
Since even for `DataElements`, you will reach a decent chunk of the "model-tree" due to references. 

These are just types and will be removed for builds. Hopefully this will be part of platform, or it's own package in the future. 

I've merged all types to a single file to reduce the number of files.

### CODE REVIEW

File: .eslintignore [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# DHIS2 Platform
```
Comment (Birkbjo) [inline .eslintignore:1]: Seems like `.gitignore` is used by default. However once I added this file, the files were not ignored anymore - so I just copied the `.gitignore` contents.

File: src/types/generated/index.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+export type * from './models'
+export type {
+    GistModel,
+    GetReferencedModels,
+    PickValue,
+    PickReferenceProperties,
+    GetReferencedModelsUnion,
+    PickInModelReferences,
+} from './utility'
```
Comment (Mohammer5) [inline src/types/generated/index.ts:11]: Why not `export * from './utility'`?

**Review by Mohammer5 — APPROVED** [review]
LGTM, just one question

File: src/types/generated/index.ts (hunk 2 above)
Comment (Birkbjo) [reply src/types/generated/index.ts:11]: I can change it to *. It's just been through some iterations. Earlier the index had named exports for each model. So I just followed that pattern. By merging to one file it was easier to just `import *`. The only benefit from named export that I can think of is that it makes it easier to `rename` a single export.

Comment (Mohammer5) [reply src/types/generated/index.ts:11]: I guess in this case it's easier to use `export *` as that will allow us to generate types again without having to touch this file

**Pull request merged**